### PR TITLE
Migrate Python code formatter from `black` to `ruff format`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132577
* #132576
* __->__ #132574

See also:

- #124845
- #123062

Closes #124845

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: .lintrunner.toml [hunk 1]
```diff
@@ -1477,6 +1477,7 @@ init_command = [
     'black==23.12.1',
     'usort==1.0.8.post1',
     'isort==5.13.2',
```
Comment (Skylion007) [inline .lintrunner.toml:1479]: Won't ruff fmt also handle isort's job too?

Comment (XuehaiPan) [reply .lintrunner.toml:1479]: `ruff check --select=I --fix` do not respect `# isort: split` for asterisk imports.

```python
from xxx.b import *  # isort: split
from xxx.a import *  # isort: split  # avoid circular imports
```

will be format to:

```python
from xxx.a import *  # isort: split  # avoid circular imports
from xxx.b import *  # isort: split
```

Comment (XuehaiPan) [reply .lintrunner.toml:1479]: FYI, https://github.com/astral-sh/ruff/issues/11776#issuecomment-2164181052

**Pull request closed without merging**